lighthouse-treemap/app/src/main.js

```diff
@@ -9,6 +9,253 @@
 
 /* globals webtreemap TreemapUtil */
 
+/** @type {TreemapViewer} */
+let treemapViewer;
+
+class TreemapViewer {
+  /**
+   * @param {LH.Treemap.Options} options
+   * @param {HTMLElement} el
+   */
+  constructor(options, el) {
+    const treemapDebugData = /** @type {LH.Audit.Details.DebugData} */ (
+      options.lhr.audits['script-treemap-data'].details);
+    if (!treemapDebugData || !treemapDebugData.treemapData) {
+      throw new Error('missing script-treemap-data');
+    }
+
+    /** @type {import('../../../lighthouse-core/audits/script-treemap-data').TreemapData} */
+    const scriptRootNodes = treemapDebugData.treemapData;
+
+    /** @type {WeakMap<LH.Treemap.Node, LH.Treemap.RootNodeContainer>} */
+    this.nodeToRootNodeMap = new WeakMap();
+
+    /** @type {{[group: string]: LH.Treemap.RootNodeContainer[]}} */
+    this.rootNodesByGroup = {
+      scripts: scriptRootNodes,
+    };
+
+    for (const rootNodes of Object.values(this.rootNodesByGroup)) {
+      for (const rootNode of rootNodes) {
+        TreemapUtil.dfs(rootNode.node, node => this.nodeToRootNodeMap.set(node, rootNode));
```

Comment:
it feels like a codesmell that we gotta set up this lookup map.  

but i don't have a great alternative just yet. it'd probably require some changes in webtreemap that violate the design.

so oh well!